mlflow/sklearn/utils.py

```diff
@@ -174,81 +203,169 @@ def _get_classifier_metrics(fitted_estimator, fit_args, fit_kwargs):
                 [
-                    _SklearnMetric(
-                        name=_METRICS_PREFIX + "roc_auc_score",
+                    _SklearnMetricOrArtifact(
+                        name=_TRAINING_PREFIX + "roc_auc_score",
                         function=sklearn.metrics.roc_auc_score,
                         arguments=dict(
                             y_true=y_true,
-                            y_score=y_pred_proba,
+                            y_score=y_pred_proba_binary,
                             average="weighted",
                             sample_weight=sample_weight,
                             multi_class="ovo",
                         ),
+                        title="",
                     ),
                 ]
             )
 
     return _get_metrics_value_dict(classifier_metrics)
 
 
+def _get_classifier_artifacts(fitted_estimator, fit_args, fit_kwargs):
+    """
+    Draw and record various common artifacts for classifier
+
+    For all classifiers, we always log:
+    (1) confusion matrix:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.plot_confusion_matrix.html
+
+    For only binary classifiers, we will log:
+    (3) precision recall curve:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.plot_precision_recall_curve.html#sklearn.metrics.plot_precision_recall_curve
+    (2) roc curve:
+    https://scikit-learn.org/stable/auto_examples/model_selection/plot_roc.html
+
+    Steps:
+    1. Extract X and y_true from fit_args and fit_kwargs, and split into train & test datasets.
+    2. If the sample_weight argument exists in fit_func (accuracy_score by default
+    has sample_weight), extract it from fit_args or fit_kwargs as
+    (y_true, y_pred, sample_weight, multioutput), otherwise as (y_true, y_pred, multioutput)
+    3. return a list of artifacts path to be logged
+
+    :param fitted_estimator: The already fitted regressor
+    :param fit_args: Positional arguments given to fit_func.
+    :param fit_kwargs: Keyword arguments given to fit_func.
+    :return: List of artifacts to be logged
+    """
+    import sklearn
+
+    fit_arg_names = _get_arg_names(fitted_estimator.fit)
+    X, y_true = _get_samples_and_labels(fit_args, fit_kwargs, fit_arg_names)
+    y_pred = fitted_estimator.predict(X)
+    sample_weight = (
+        _get_sample_weight(fit_arg_names, fit_args, fit_kwargs)
+        if _SAMPLE_WEIGHT in fit_arg_names
+        else None
+    )
+    labels = _get_labels(y_true)
+    classifier_artifacts = [
+        _SklearnMetricOrArtifact(
+            name=_TRAINING_PREFIX + "confusion_matrix",
+            function=sklearn.metrics.plot_confusion_matrix,
+            arguments=dict(
+                estimator=fitted_estimator,
+                X=X,
+                y_true=y_pred,
+                sample_weight=sample_weight,
+                normalize="true",
+                labels=labels,
+            ),
+            title="Confusion Matrix",
```

Comment:
Nit: Each word in this title is capitalized, while the precision / recall curve plot titles are not. We should be consistent. My vote would be to only capitalize the first letter of the first word in the title; this is consistent with the axis labeling convention in sklearn.